packages/blockchain-link/src/workers/solana/index.ts

```diff
@@ -70,30 +105,21 @@ const getAllSignatures = async (
 
 const fetchTransactionPage = async (
     api: SolanaAPI,
-    signatures: string[],
+    signatures: Signature[],
 ): Promise<ParsedTransactionWithMeta[]> => {
-    // avoid requests that are too big by querying max N signatures at once
-    const perChunk = 50; // items per chunk
-    const confirmedSignatureChunks = signatures.reduce((resultArray, item, index) => {
-        const chunkIndex = Math.floor(index / perChunk);
-        if (!resultArray[chunkIndex]) {
-            resultArray[chunkIndex] = []; // start a new chunk
-        }
-        resultArray[chunkIndex].push(item);
-
-        return resultArray;
-    }, [] as string[][]);
-
-    const confirmedTxsChunks = await Promise.all(
-        confirmedSignatureChunks.map(signatureChunk =>
-            api.getParsedTransactions(signatureChunk, {
-                maxSupportedTransactionVersion: 0,
-                commitment: 'confirmed',
-            }),
-        ),
-    );
-
-    return confirmedTxsChunks.flat().filter((tx): tx is ParsedTransactionWithMeta => !!tx);
+    return (
+        await Promise.all(
+            signatures.map(signature =>
+                api.rpc
+                    .getTransaction(signature, {
+                        encoding: 'jsonParsed',
+                        maxSupportedTransactionVersion: 0,
+                        commitment: 'confirmed',
+                    })
+                    .send(),
+            ),
+        )
```

Comment:
RPC providers generally hate it when we make request batches. It thwarts their ability to load balance, and it creates an all-or-nothing dynamic where one failed request invalidates the entire batch. Typically HTTP/2 transports give us all the performance of manual batching, and all of the benefits of routing on the backend. You may like to visit the call to `createDefaultRpcTransport()` and experiment with the `dispatcher_NODE_ONLY` config option.